task/buildah-remote-oci-ta/0.4/buildah-remote-oci-ta.yaml

```diff
@@ -829,8 +825,24 @@ spec:
 
       echo "Running syft on the source directory"
       syft dir:"/var/workdir/$SOURCE_CODE_DIR/$CONTEXT" --output "$syft_sbom_type"="/var/workdir/sbom-source.json"
+
+      cat >./script-sbom-image.sh <<'SBOMIMAGEEOF'
+      #!/bin/bash
+      set -euo pipefail
+      buildah mount $(cat /shared/container_name)
+      # delete symlinks - they may point outside the container rootfs, messing with SBOM scanners
+      find $(cat /shared/container_path) -xtype l -delete
       echo "Running syft on the image filesystem"
       syft dir:"$(cat /shared/container_path)" --output "$syft_sbom_type"="/var/workdir/sbom-image.json"
+      SBOMIMAGEEOF
+      chmod a+x ./script-sbom-image.sh
+
+      unshare -Uf --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 --mount ./script-sbom-image.sh
+    securityContext:
+      runAsUser: 0
```

Comment:
why do we need root?

Reply:
+1, @zxiong could you add more info about the changes in the commit messages?